### PR TITLE
Update controller-runtime dependency

### PR DESCRIPTION
**Pull request by Diaphteiros**
**How to categorize this PR?**

/area landscaper
/kind enhancement
/priority blocker

**What this PR does / why we need it**:
Upgrades the controller-runtime dependency to `v0.8.1`.
Also updates the k8s dependency in apis/go.mod

**Special notes for your reviewer**:

**Release note**:

```other dependency
Update controller-runtime dependency to `v0.8.1`
```

### CODE REVIEW

**Comment by Diaphteiros** [comment]
The problem is probably that with `github.com/gardener/landscaper/apis v0.0.0` in the main go.mod file, the wrong version of the apis is vendored. Not sure how to resolve this, since I would need to vendor the version that I'm currently working on ...

File: pkg/deployer/helm/ensure.go [hunk 1]
```diff
@@ -201,16 +201,17 @@ func (h *Helm) DeleteFiles(ctx context.Context) error {
...
-	controllerutil.RemoveFinalizer(&h.DeployItem.ObjectMeta, lsv1alpha1.LandscaperFinalizer)
+	controllerutil.RemoveFinalizer(h.DeployItem, lsv1alpha1.LandscaperFinalizer)
 	return h.kubeClient.Update(ctx, h.DeployItem)
 }
 
 // ApplyObject applies a managed resource to the target cluster.
 func (h *Helm) ApplyObject(ctx context.Context, kubeClient client.Client, obj *unstructured.Unstructured) error {
 	currOp := "ApplyObjects"
-	currObj := obj.NewEmptyInstance()
+	currObj := unstructured.Unstructured{}
```
Comment (schrodit) [inline pkg/deployer/helm/ensure.go:211]: has `NewEmptyInstance` been removed?

**Review by schrodit — CHANGES_REQUESTED** [review]
looks good when the test is fixed.
Can you directly do a refactoring for that todo now that the new clinet.Object is available: https://github.com/gardener/landscaper/blob/master/test/utils/envtest/types.go#L20

File: pkg/deployer/helm/ensure.go (hunk 1 above)
Comment (Diaphteiros) [reply pkg/deployer/helm/ensure.go:211]: No, but it returns an object of type `runtime.Unstructured` which does not implement the `client.Object` interface that is now required by `kubeClient.Get`. I'm somewhat confused why `NewEmptyInstance` called on type `unstructured.Unstructured` - which implements the interface - does return a different type. This was the first workaround that I found - it basically does exactly the same thing as `NewEmptyInstance`, but with the `unstructured.Unstructured` type.

Comment (schrodit) [reply pkg/deployer/helm/ensure.go:211]: ah makes sense.
Can you add that as a comment maybe

Comment (Diaphteiros) [reply pkg/deployer/helm/ensure.go]: done

**Comment by Diaphteiros** [comment]
> looks good when the test is fixed.
> Can you directly do a refactoring for that todo now that the new clinet.Object is available: https://github.com/gardener/landscaper/blob/master/test/utils/envtest/types.go#L20

done

**Review by schrodit — APPROVED** [review]
looks good now 
/lgtm

**Pull request merged**